src/Domain/Stack/LiipTransformerStackExecutor.php

```diff
@@ -0,0 +1,186 @@
+use Liip\ImagineBundle\Domain\ImageReferenceFile;
+use Liip\ImagineBundle\Domain\Imagine\Filter\FilterExecutor;
+use Liip\ImagineBundle\Domain\PostProcessor\PostProcessor;
+use Liip\ImagineBundle\Imagine\Filter\FilterConfiguration;
+
+class LiipTransformerStackExecutor implements TransformerStackExecutor
+{
+    /**
+     * @var array<string, FilterExecutor>
+     */
+    private array $filters = [];
+
+    /**
+     * @var array<string, PostProcessor>
+     */
+    private array $postProcessors = [];
+
+    public function __construct(
+        private FilterConfiguration $filterConfiguration,
+        private ImagineInterface $imagine,
+        private MimeTypeGuesserInterface $mimeTypeGuesser,
+    ) {}
+
+    /**
+     * Adds a loader to handle the given filter.
+     */
+    public function addFilter(string $filterName, FilterExecutor $executor): void
+    {
+        $this->filters[$filterName] = $executor;
+    }
+
+    /**
+     * Adds a post-processor to handle binaries.
+     */
+    public function addPostProcessor(string $name, PostProcessor $postProcessor): void
+    {
+        $this->postProcessors[$name] = $postProcessor;
+    }
+
+    /**
+     * Apply the stack to the image.
+     */
+    public function apply(string $stackName, ImageReference $sourceImageReference): ImageReference
+    {
+        $config = $this->filterConfiguration->get($stackName);
+
+        $config += [
+            'quality' => 100,
+            'animated' => false,
+        ];
+
+        if ($sourceImageReference instanceof ImageReferenceFile) {
+            $image = $this->imagine->open($sourceImageReference->getPath());
+        } else {
+            $image = $this->imagine->load($sourceImageReference->getContent());
+        }
+
+        $image = $this->applyFilters($image, $config);
+        $resultImageReference = $this->exportConfiguredImageBinary($sourceImageReference, $image, $config);
+
+        return $this->applyPostProcessors($resultImageReference, $config);
+    }
+
+    public function applyFilters(ImageInterface $image, array $config): ImageInterface
+    {
+        foreach ($this->sanitizeFilters($config['filters'] ?? []) as $name => $options) {
+            $prior = $image;
+            $image = $this->filters[$name]->load($image, $options);
+
+            if ($prior !== $image) {
+                $this->destroyImage($prior);
+            }
+        }
+
+        return $image;
+    }
+
+    public function applyPostProcessors(ImageReference $image, array $config): ImageReference
+    {
+        foreach ($this->sanitizePostProcessors($config['post_processors'] ?? []) as $name => $options) {
+            $image = $this->postProcessors[$name]->process($image, $options);
+        }
+
+        return $image;
+    }
+
+    private function exportConfiguredImageBinary(ImageReference $imageReference, ImageInterface $image, array $config): ImageReference
+    {
+        // TODO: this is for now literal copy-paste from FilterManager
```

Comment:
please challenge the things below, but i don't know if we can do much better. i'd not change the configuration if we don't really need to.